network/src/drop_join.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use tokio::sync::mpsc;
+
+pub trait DropJoinable: Send + Sync + 'static {
+    fn destroy(self);
+}
+
+impl<X: Send + Sync + 'static> DropJoinable for tokio::task::JoinHandle<X> {
+    fn destroy(self) {
+        self.abort();
+    }
+}
+
+impl<X: Send + Sync + 'static> DropJoinable for std::thread::JoinHandle<X> {
+    fn destroy(self) {
+        // todo: this could be blocking in an async thread??
```

Comment:
it can be; it would probably be a good idea to launch a `tokio::task::spawn_blocking` over this operation

Reply:
agreed, done